app/jinja_filters.py

```diff
@@ -253,15 +253,26 @@ def should_wrap_with_fieldset_processor() -> dict[str, Callable]:
     return {"should_wrap_with_fieldset": should_wrap_with_fieldset}
 
 
+def get_min_max_values(
+    min_max: Literal["minimum", "maximum"], answer: AnswerType, default_value: int
+) -> int:
+    schema = g.get("schema")
+
+    if answer.get(min_max, {}) and isinstance(answer[min_max]["value"], dict):
+        if (
+            identifier := answer[min_max]["value"].get("identifier", None)
+        ) in schema.min_and_max_map:
+            return len(str(schema.min_and_max_map[identifier]))
+
+    return len(str(answer.get(min_max, {}).get("value", default_value)))
+
+
 @blueprint.app_template_filter()
 def get_width_for_number(answer: AnswerType) -> Optional[int]:
     allowable_widths = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 15, 20, 30, 40, 50]
 
-    min_value = answer.get("minimum", {}).get("value", 0)
-    max_value = answer.get("maximum", {}).get("value", MAX_NUMBER)
-
-    min_value_width = len(str(min_value))
-    max_value_width = len(str(max_value))
+    min_value_width = get_min_max_values("minimum", answer, 0)
+    max_value_width = get_min_max_values("maximum", answer, MAX_NUMBER)
 
     width = min_value_width if min_value_width > max_value_width else max_value_width
```

Comment:
Just a suggestion, since this isn't part of the change. Could we use  `width = max(min_value_width, max_value_width)` instead. Makes it a bit clearer for me, but I don't mind if you prefer the if statement.